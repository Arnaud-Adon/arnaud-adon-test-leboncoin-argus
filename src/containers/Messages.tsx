import React, { FunctionComponent, useEffect, useState } from "react";
import MessageForm from "../components/Form/MessageForm";
import MessagesList from "../components/List/MessagesList";
import { IMessage } from "../models/message";
import { getMessages } from "../services/api";
import "./Message.css";

const Messages: FunctionComponent = () => {
  const [messages, setMessages] = useState<IMessage[]>([]);

  useEffect(() => {
    getMessages().then((messages) => setMessages(messages));
  }, []);

  const addMessage = (message: IMessage): void => {
    setMessages((prevMessages) => [...prevMessages, message]);
  };

  return (
    <div className="container">
      <MessagesList messagesList={messages} />
      <MessageForm messages={messages} addMessage={addMessage} />
    </div>
  );
};

export default Messages;
