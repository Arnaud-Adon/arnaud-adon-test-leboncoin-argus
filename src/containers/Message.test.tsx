import React from "react";
import { render, act } from "@testing-library/react";
import Messages from "./Messages";

describe("Message container test suite", () => {
  it("should render correctly", async () => {
    const wrapper = render(<Messages />);
    await act(() => Promise.resolve());
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain a container", async () => {
    const { container } = render(<Messages />);
    expect(container.firstChild).toHaveClass("container");
    await act(() => Promise.resolve());
  });
});
