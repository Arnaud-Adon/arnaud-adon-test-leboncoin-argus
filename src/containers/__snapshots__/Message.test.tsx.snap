// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message container test suite should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="messages-container"
        >
          <div
            class="message-content left-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content left-color"
            >
              Bonjour à tous
            </div>
          </div>
          <div
            class="message-content right-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content right-color"
            >
              Comment allez-vous
            </div>
          </div>
          <div
            class="message-content left-message"
          >
            <p
              class="message-field"
            >
              private
            </p>
            <div
              class="message-text-content left-color"
            >
              Voici un secret
            </div>
          </div>
          <div
            class="message-content right-message"
          >
            <p
              class="message-field"
            >
              private
            </p>
            <div
              class="message-text-content right-color"
            >
              Ok, je garde ça pour moi
            </div>
          </div>
          <div
            class="message-content left-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content left-color"
            >
              Salut
            </div>
          </div>
          <div
            class="message-content right-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content right-color"
            >
              La forme ?
            </div>
          </div>
          <div
            class="message-content left-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content left-color"
            >
              Bienvenue
            </div>
          </div>
          <div
            class="message-content right-message"
          >
            <p
              class="message-field"
            >
              public
            </p>
            <div
              class="message-text-content right-color"
            >
              Merci à toi.
            </div>
          </div>
        </div>
        <div
          class="form-message"
        >
          <label
            for="form-message"
          >
            Entrez un message
          </label>
          <form>
            <textarea
              cols="30"
              id="form-message"
              name="message"
              placeholder="Entrez votre message"
              rows="10"
            />
            <div>
              <label
                for="message-status"
              >
                Message en privé ?
              </label>
              <input
                id="message-status"
                type="checkbox"
              />
            </div>
            <button
              class="message-button disabled"
              disabled=""
            >
              Valider
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="messages-container"
      >
        <div
          class="message-content left-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content left-color"
          >
            Bonjour à tous
          </div>
        </div>
        <div
          class="message-content right-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content right-color"
          >
            Comment allez-vous
          </div>
        </div>
        <div
          class="message-content left-message"
        >
          <p
            class="message-field"
          >
            private
          </p>
          <div
            class="message-text-content left-color"
          >
            Voici un secret
          </div>
        </div>
        <div
          class="message-content right-message"
        >
          <p
            class="message-field"
          >
            private
          </p>
          <div
            class="message-text-content right-color"
          >
            Ok, je garde ça pour moi
          </div>
        </div>
        <div
          class="message-content left-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content left-color"
          >
            Salut
          </div>
        </div>
        <div
          class="message-content right-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content right-color"
          >
            La forme ?
          </div>
        </div>
        <div
          class="message-content left-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content left-color"
          >
            Bienvenue
          </div>
        </div>
        <div
          class="message-content right-message"
        >
          <p
            class="message-field"
          >
            public
          </p>
          <div
            class="message-text-content right-color"
          >
            Merci à toi.
          </div>
        </div>
      </div>
      <div
        class="form-message"
      >
        <label
          for="form-message"
        >
          Entrez un message
        </label>
        <form>
          <textarea
            cols="30"
            id="form-message"
            name="message"
            placeholder="Entrez votre message"
            rows="10"
          />
          <div>
            <label
              for="message-status"
            >
              Message en privé ?
            </label>
            <input
              id="message-status"
              type="checkbox"
            />
          </div>
          <button
            class="message-button disabled"
            disabled=""
          >
            Valider
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
